### PR TITLE
Add credentials store support

### PR DESCRIPTION
**Pull request by shin-**

### CODE REVIEW

**Comment by shin-** [comment]
https://github.com/shin-/dockerpy-creds

**Comment by shin-** [comment]
Fixes #1023

File: docker/auth/auth.py [hunk 1]
```diff
@@ -105,6 +114,34 @@ def resolve_authconfig(authconfig, registry=None):
...
+            res['IdentityToken'] = data['Secret']
+        else:
+            res.update({
+                'Username': data['Username'],
+                'Password': data['Secret'],
+            })
+        return res
+    except dockerpycreds.StoreError as e:
+        log.error('Credentials store error: {0}'.format(repr(e)))
+
```
Comment (dnephin) [inline docker/auth/auth.py]: I wonder if this should raise an exception? The user explicitly asked to use the credential store, right ?

**Comment by dnephin** [comment]
LGTM, just one comment.

I'm not all that familiar with these credential stores, but the code looks good to me.

File: docker/auth/auth.py (hunk 1 above)
Comment (shin-) [inline docker/auth/auth.py]: Yeah, I need to differentiate a runtime error from a missing entry in dockerpycreds, then we can raise when there's an issue, and proceed with an unauthenticated push/pull if we just don't have credentials.

**Pull request merged**